beamer/chain.py

```diff
@@ -46,7 +46,7 @@ class EventMonitor:
     def __init__(
         self,
         web3: Web3,
-        contracts: tuple[Contract, ...],
+        contracts: tuple[Contract | AsyncContract, ...],
```

Comment:
Can we use just `BaseContract` from `web3.contract` here instead of a union of `Contract` and `AsyncContract`?
Those two anyway inherit from `BaseContract` and I don't think we need to differentiate between them anywhere. It would make typing less tedious.

Reply:
will try

Reply:
BaseContract has no .functions, it has .all_functions instead but when i tried it didnt work anyway. I guess i need to cast returned contract from w3.eth.contract to be a `Contract` and not AsyncContract

Reply:
casted to Contract. should be better.